### PR TITLE
Add FileSystem Events Example

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: examples/FileSystemEvent.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+watchPaths :: NonEmpty (Array Word8) -> SerialT IO Event.Event
+watchPaths = Event.watchTrees
+
+main :: IO ()
+main = do
+    args <- getArgs
+    paths <- mapM toUtf8 args
+
+    watchPaths (NonEmpty.fromList paths)
```
Comment (harendra-kumar) [inline examples/FileSystemEvent.hs:45]: We can directly use `watchTrees` here instead of creating the `watchPaths` function.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Comment by rnjtranjan** [comment]
Fixes #6

**Comment by harendra-kumar** [comment]
GHC-8.8 build is failing with a compilation error.

**Comment by harendra-kumar** [comment]
We can remove the x86 build from CI.

**Comment by harendra-kumar** [comment]
Need to fix ghcjs build. Even though the network package compiles with ghcjs we cannot really use it, need to disable some examples for ghcjs as before.

**Pull request merged**